### PR TITLE
Backbone for UI App

### PR DESCRIPTION
**Pull request by lucasponce**
This PR contains:
- Extract UI as a separate project.
- Setup all automation for UI based on npm, bower and grunt tool.
- Add UI build into pom.xml.
- Basic mockup based on AngularJS and PatternFly.

### CODE REVIEW

**Comment by lucasponce** [comment]
The mock up has only intention to enable all PatternFly + AngularJS technology to speedup the learning curve and focus on features instead to spend time defining layouts and conventions.

image

image

image

File: ui/README.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+```
+    http://<host>:<port>/hawkular/alerts/ui
+```
+
+== Working on UI
+
+To develop in the UI Application you will need NodeJS tools.
+
+A standard strategy could be to work locally on your
```
Comment (jmazzitelli) [inline ui/README.adoc]: You should make it clear that the developer should "cd" to the directory (.../src/main/ui) before running the npm/bower commands below.

File: ui/README.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+directory and prepare the dependencies you need to run all the UI elements
+
+```
+    npm install
+    sudo npm install -g bower
+    bower install
+    sudo npm install -g grunt-cli
+    sudo npm install -g compass
+    gem install sass
```
Comment (jmazzitelli) [inline ui/README.adoc]: "gem" -- this requires Ruby / RVM to be installed I assume?

File: ui/README.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    gem install compass
+```
+
+These steps will create some folders that will be part of the .gitignore as they can't be pushed into the repository
```
Comment (jmazzitelli) [inline ui/README.adoc]: After these instructions, I have the following that is not in .gitignore - git wants me to commit it. I think you will want to put this in .gitignore, too, correct?

`ui/src/main/ui/package-lock.json`

File: ui/README.adoc (hunk 1 above)
Comment (lucasponce) [inline ui/README.adoc]: +1

File: ui/README.adoc (hunk 2 above)
Comment (lucasponce) [inline ui/README.adoc]: Yes. I don't pretend to be comprehensive here, basically we are pointing to AngularJS + PatternFly and re-using the demo app, so it is possible that some additional note should be added in following iterations.

File: ui/README.adoc (hunk 3 above)
Comment (lucasponce) [inline ui/README.adoc]: Yes, it is weird, I didn't have it on my machine.
But yes, the lock shouldn't be pushed, so +1 to add it into the .gitignore.

**Comment by lucasponce** [comment]
Travis worked fine, did you get able to have a local environment or is there any important step missed ?

**Comment by jmazzitelli** [comment]
I have it running and I can see it in my browser when `grunt server` is run.

**Comment by lucasponce** [comment]
Cool, thanks @jmazzitelli, going to add your comments and merge it.

**Pull request merged**